ParquetSharp.Dataset/Filter/TimestampRangeStatisticsEvaluator.cs

```diff
@@ -0,0 +1,33 @@
+using System;
+
+namespace ParquetSharp.Dataset.Filter;
+
+internal sealed class TimestampRangeStatisticsEvaluator
+    : ILogicalStatisticsVisitor<DateTime, bool>
+        , ILogicalStatisticsVisitor<DateTimeNanos, bool>
+{
+    public TimestampRangeStatisticsEvaluator(DateTime start, DateTime end)
+    {
+        _start = start;
+        _end = end;
+    }
+
+    public bool Visit(LogicalStatistics stats)
+    {
+        return true;
+    }
+
+    public bool Visit(LogicalStatistics<DateTime> stats)
+    {
+        return _end > stats.Min && _start <= stats.Max;
```

Comment:
I was wondering whether this condition is correct, but looked at the tests and yeah, it is all good😄

Reply:
Yeah I had to draw a diagram to convince myself 😄